### PR TITLE
py3-sphinx: Fix dependencies for sphinx, add rtd-theme and sphinx-7 

### LINKED ISSUE #26703: py3-sphinx missing its dependencies
```
77d032b47a79:/work/packages# apk add py3-sphinx
77d032b47a79:/work/packages# sphinx-build --version
Traceback (most recent call last):
  File "/usr/bin/sphinx-build", line 5, in <module>
    from sphinx.cmd.build import main
  File "/usr/lib/python3.12/site-packages/sphinx/cmd/build.py", line 17, in <module>
    from docutils.utils import SystemMessage
ModuleNotFoundError: No module named 'docutils'
```

### PR DESCRIPTION
**Pull request by pnasrat**
Fixes: #26703

Some major sphinx themes such as for readthedocs and jupyterbooks do not support sphinx 8 yet

See https://github.com/readthedocs/sphinx_rtd_theme/issues/1582

- Adds py3-sphinx-7
- Adds py3-sphinx-rtd-theme

Signed-off-by: Pris Nasrat <pris.nasrat@chainguard.dev>

### CODE REVIEW

**Comment by pnasrat** [comment]
Per the issue circular deps would have been in place however the core sphinx plugins do not requires sphinx this makes our packaging consistent with pypi 

```console
 ~/scratch/sphinx-deps $ pip install --dry-run --report r.json sphinxcontrib-applehelp sphinxcontrib-devhelp sphinxcontrib-htmlhelp sphinxcontrib-jsmath sphinxcontrib-qthelp sphinxcontrib-serializinghtml
Collecting sphinxcontrib-applehelp
  Using cached sphinxcontrib_applehelp-2.0.0-py3-none-any.whl.metadata (2.3 kB)
Collecting sphinxcontrib-devhelp
  Using cached sphinxcontrib_devhelp-2.0.0-py3-none-any.whl.metadata (2.3 kB)
Collecting sphinxcontrib-htmlhelp
  Using cached sphinxcontrib_htmlhelp-2.1.0-py3-none-any.whl.metadata (2.3 kB)
Collecting sphinxcontrib-jsmath
  Using cached sphinxcontrib_jsmath-1.0.1-py2.py3-none-any.whl.metadata (1.4 kB)
Collecting sphinxcontrib-qthelp
  Using cached sphinxcontrib_qthelp-2.0.0-py3-none-any.whl.metadata (2.3 kB)
Collecting sphinxcontrib-serializinghtml
  Using cached sphinxcontrib_serializinghtml-2.0.0-py3-none-any.whl.metadata (2.4 kB)
Using cached sphinxcontrib_applehelp-2.0.0-py3-none-any.whl (119 kB)
Using cached sphinxcontrib_devhelp-2.0.0-py3-none-any.whl (82 kB)
Using cached sphinxcontrib_htmlhelp-2.1.0-py3-none-any.whl (98 kB)
Using cached sphinxcontrib_jsmath-1.0.1-py2.py3-none-any.whl (5.1 kB)
Using cached sphinxcontrib_qthelp-2.0.0-py3-none-any.whl (88 kB)
Using cached sphinxcontrib_serializinghtml-2.0.0-py3-none-any.whl (92 kB)
Would install sphinxcontrib-applehelp-2.0.0 sphinxcontrib-devhelp-2.0.0 sphinxcontrib-htmlhelp-2.1.0 sphinxcontrib-jsmath-1.0.1 sphinxcontrib-qthelp-2.0.0 sphinxcontrib-serializinghtml-2.0.0
```

**Review by smoser — CHANGES_REQUESTED** [review]
Some questions.
None of it significant.

thanks for the individual commits.

File: py3-commonmark.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  name: py3-commonmark
+  version: 0.9.1
+  epoch: 0
+  description: Python parser for the CommonMark Markdown spec
+  copyright:
+    - license: BSD-3-Clause
+  dependencies:
+    runtime:
+      - py3-future
+      - python-3
```
[the remark this answers is not in the document]
Comment (pnasrat) [reply py3-commonmark.yaml:12]: no it was generated

File: py3-sphinxcontrib-jquery.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+pipeline:
+  - uses: fetch
+    with:
+      expected-sha256: 1620739f04e36a2c779f1a131a2dfd49b2fd07351bf1968ced074365933abc7a
+      uri: https://files.pythonhosted.org/packages/source/s/sphinxcontrib-jquery/sphinxcontrib-jquery-${{package.version}}.tar.gz
+
+  - name: Python Build
+    uses: python/build-wheel
+
+  - uses: strip
```
[the remark this answers is not in the document]
Comment (pnasrat) [reply py3-sphinxcontrib-jquery.yaml:32]: I swear I added that will push

File: py3-recommonmark.yaml [hunk 3]
```diff
@@ -35,6 +35,9 @@ pipeline:
 
 update:
   enabled: true
+  ignore-regex-patterns:
+    - (.*)(a|b|rc)(.*) # Ignore alpha, beta, and rc releases
```
[the remark this answers is not in the document]
Comment (pnasrat) [reply py3-recommonmark.yaml]: the update check didn't handle all those a while back yeah it doesn't handle the a and b that python use 

```
"alpha", "beta", "rc", "pre"
```

File: py3-recommonmark.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  epoch: 0
+  description: A docutils-compatibility bridge to CommonMark, enabling you to write CommonMark inside of Docutils & Sphinx projects.
+  copyright:
+    - license: MIT
+  dependencies:
+    runtime:
+      - py3-commonmark
+      - py3-docutils
+      - py3-sphinx
+      - python-3
```
Comment (smoser) [inline py3-recommonmark.yaml:14]: drop this?

File: py3-sphinx-rtd-theme.yaml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  epoch: 0
+  description: Read the Docs theme for Sphinx
+  copyright:
+    - license: MIT
+  dependencies:
+    runtime:
+      - py3-sphinx-7
+      - py3-docutils
+      - py3-sphinxcontrib-jquery
+      - python-3
```
Comment (smoser) [inline py3-sphinx-rtd-theme.yaml:14]: this

File: py3-sphinxcontrib-jquery.yaml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  name: py3-sphinxcontrib-jquery
+  version: "4.1"
+  epoch: 0
+  description: Extension to include jQuery on newer Sphinx releases
+  copyright:
+    - license: 0BSD
+  dependencies:
+    runtime:
+      - py3-sphinx
+      - python-3
```
Comment (smoser) [inline py3-sphinxcontrib-jquery.yaml:12]: this

File: py3-sphinx-rtd-theme.yaml [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  - name: Python Build
+    uses: python/build-wheel
+
+  - uses: strip
+
+update:
+  enabled: true
+  manual: false
+  ignore-regex-patterns:
+    - (.*)(a|b|rc)(.*) # Ignore alpha, beta, and rc releases
```
Comment (smoser) [inline py3-sphinx-rtd-theme.yaml:40]: i don't think this is necessary. I just ran `wolfictl check-update` against the file without lines 39 and 40 and it did not suggest there was an update (there is a 3.0.0rc1 tag in that repo https://github.com/readthedocs/sphinx_rtd_theme/tags )

File: py3-sphinxcontrib-devhelp.yaml [hunk 8]
```diff
@@ -1,13 +1,12 @@
...
   version: 2.0.0
-  epoch: 0
+  epoch: 1
   description: sphinxcontrib-devhelp is a sphinx extension which outputs Devhelp documents
   copyright:
     - license: BSD-2-Clause
   dependencies:
     runtime:
-      - py3-sphinx
       - python3
```
Comment (smoser) [inline py3-sphinxcontrib-devhelp.yaml:10]: drop?

File: py3-sphinxcontrib-serializinghtml.yaml [hunk 9]
```diff
@@ -1,13 +1,12 @@
...
   version: 2.0.0
-  epoch: 0
+  epoch: 1
   description: sphinxcontrib-serializinghtml is a sphinx extension which outputs "serialized" HTML files (json and pickle)
   copyright:
     - license: BSD-2-Clause
   dependencies:
     runtime:
-      - py3-sphinx
       - python3
```
Comment (smoser) [inline py3-sphinxcontrib-serializinghtml.yaml:10]: drop?

**Review by smoser — CHANGES_REQUESTED** [review]

**Review by smoser — APPROVED** [review]
thanks.

**Pull request merged**